test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/DrainModeResumeEndToEndTests.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.WebJobs.Script.Tests;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.EndToEnd
+{
+    [Trait(TestTraits.Category, TestTraits.EndToEnd)]
+    [Trait(TestTraits.Group, TestTraits.DrainModeEndToEnd)]
+    public class DrainModeResumeEndToEndTests : ResumeTestFixture
+    {
+        [Fact]
+        public async Task DrainModeEnabled_RunningHost_StartsNewHost_ReturnsOk()
+        {
+            // Validate the drain state is "Disabled" initially
+            var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+            var responseString = response.Content.ReadAsStringAsync().Result;
+            var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+            Assert.Equal(status.State, DrainModeState.Disabled);
+
+            ManualResetEvent resetEvent = new ManualResetEvent(false);
+            _ = Task.Run(async () =>
+            {
+                // Put the host in drain mode
+                response = await SamplesTestHelpers.InvokeDrain(this);
+
+                // Validate the drain state is changed to "InProgress"
+                await TestHelpers.Await(async () =>
+                {
+                    var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+                    var responseString = response.Content.ReadAsStringAsync().Result;
+                    var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+                    return status.State == DrainModeState.InProgress;
+                }, 20000);
+
+                // Validate host is "Running" after resume is called
+                await TestHelpers.Await(async () =>
+                {
+                    var response = await SamplesTestHelpers.InvokeResume(this);
+                    var responseString = response.Content.ReadAsStringAsync().Result;
+                    var status = JObject.Parse(responseString);
+
+                    return (string)status["hostStatus"] == ScriptHostState.Running.ToString();
+                }, 20000);
+
+                // Validate the drain state is changed to "Disabled"
+                await TestHelpers.Await(async () =>
+                {
+                    var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+                    var responseString = response.Content.ReadAsStringAsync().Result;
+                    var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+                    return status.State == DrainModeState.Disabled;
+                }, 20000);
+
+                resetEvent.Set();
+            });
+
+            response = await SamplesTestHelpers.InvokeHttpTrigger(this, "HttpTrigger");
```

Comment:
Is the intent here to have this function run before drain mode is initiated? Might make the call before we initiate then, and also verify it runs after resume as well.

Reply:
Yup, I think you make a good point - I want to verify it works before and after. I'll update the test!